### ISSUE TITLE
MsSQL double precision round to int

### ISSUE THREAD

**Issue by jmai00** [typeorm/typeorm#352]
I'm having issue figuring out why in MSSQL the number 1.234567789 precision is dropping to just 1 in DB while my table data type is a real. While in mysql it works fine

**Comment by jmai00** [comment]
I think I found the issue.
In line 202 of SqlServerQueryRunner, no type is passed because it's optional:
```typescript
request.input(index, parameters![index]);
```

In the node-mssql document:
if no type is set, the driver will infer the type. Which in my case was incorrect. So I'd have to ask 
Is there a possibility to send the column type info into the input parameter?

**Comment by pleerock** [comment]
no, you can't. Thats weird, why they do so? Other drivers does not require extra type determination and work on their own. Can you research is there any other way how this may work in mssql? I imagine they could just determine type from column type if needed, or at least provide special value wrapper classes like `.input(name, SqlDataTypes.REAL(10.2))`

**Comment by jmai00** [comment]
there's a way to set the type when you set the input. 
```
request.input(column_name, column_type[optional], value);

request('field1, sql.Decimal(5,3), 0.123)
```

BTW, i just found out why comment out decimal precision and scale. It doesn't work on creation probably due to table creation timing.

**Comment by pleerock** [comment]
> request('field1, sql.Decimal(5,3), 0.123) 

Yes I saw that however it wont work. It can work only if `queryBuilder.setParameter` accept three parameters. I would like to avoid third parameter because its only mssql-specific. Would be cool if mssql support it different way, like: `request('field1, sql.Decimal(0.123, 5,3)) ` and we could do then `qb.setParameter("field1", sql.Decimal(0.123, 5,3))`

**Comment by jmai00** [comment]
I think your suggestion might have to use the sql wrapper. I don't want to hack things such as this:
```
if ((typeof(parameter) == "number") && (parameter % 1 !== 0)){
                        let n = (parameter + "").split(".");
                        request.input(index, _this.driver.mssql.Decimal ( n[0].length + n[1].length + 1, n[1].length), parameters[index]);
                    }
                    else {
                        request.input(index, parameters[index]);
                    }
```

**Comment by pleerock** [comment]
I suggest you to find any good way how to solve this issue. If you wont fine anything we can ask node-mssql community to create (or most probably send them PRs if they accept this idea) data wrappers to use `request('field1, sql.Decimal(0.123, 5,3))` instead of `request('field1, sql.Decimal(5,3), 0.123)`. If they won't accept this idea, then we are forced to either add extra parameter to `QueryBuilder#setParameter` which will be ugly because it will be mssql-specific, either create our own data-wrappers which will be also ugly and extra code in the codebase. Alternatively we can try to use `tedious` driver instead of `mssql` and check if that will solve this problem and won't introduce new problems for us.

**Comment by jmai00** [comment]
I'm using tedious from the inside node-mssql because I'm running on linux. From the tedious document, you are required to pass in the data type.
```
request.addParameter(name, type, value, [options])
```
Just found out tedious supports up to 19 precision places for decimals

**Comment by pleerock** [comment]
can you research how other orms handle it, for example sequalize? As I told we can add extra parameter to `QueryBuilder#setParameter(name, value, type?)` however I don't like this approach because its mysql-specific. We should find something better.

**Comment by pleerock** [comment]
hey @jmai00 did you came up with something?

**Comment by jmai00** [comment]
Sorry now as of yet. I'm currently swap on a project at the moment, so I haven't had time to look deeply into this situation.

**Comment by jbrassard** [comment]
I have to exact same problem

**Comment by pleerock** [comment]
This issue is fixed in `0.1.0-alpha.29`. Now you can use following syntax:

```ts
createQueryBuilder("post")
   .where("post.someRealNumber = :someRealNumber", { someRealNumber: new MssqlParameter(1.234567789, "float") })
   .getMany();
```

So you basically wrapping your values (and specify its type) into special `MssqlParameter` object.

See [this test](https://github.com/pleerock/typeorm/blob/53996c40b0e4acc289b82e06efc02b7b8f712a11/test/github-issues/352/issue-352.ts#L17-L17) for confirmation.
